rspace/src/test/scala/coop/rchain/rspace/nextgenrspace/history/HistoryTest.scala

```diff
@@ -0,0 +1,114 @@
+package coop.rchain.rspace.nextgenrspace.history
+
+import coop.rchain.rspace.test.TestKey32
+import monix.eval.Task
+import org.scalatest.{Assertion, FlatSpec, Matchers, OptionValues}
+
+import scala.concurrent.duration._
+import monix.execution.Scheduler.Implicits.global
+import TestData._
+import coop.rchain.rspace.Blake2b256Hash
+import coop.rchain.rspace.nextgenrspace.history.History.Trie
+import scodec.bits.ByteVector
+import History.codecTrie
+import cats.implicits._
+
+import scala.util.Random
+
+class HistoryTest extends FlatSpec with Matchers with OptionValues with InMemoryHistoryTestBase {
+  "first leaf in trie" should "create a Skip -> Leaf" in withEmptyTrie { emptyHistory =>
+    val data = insert(_zeros) :: Nil
+    for {
+      newHistory <- emptyHistory.process(data)
+      result     <- newHistory.traverseTrie(_zeros)
```

Comment:
it's package private. it's part of the internal solution and not the external api.

Reply:
How do you get anything out of the history though? I'd expect there to be at least a `get(key): Trie` method - but maybe I'm wrong :)

Also, I'm so brain-dead after all the traverses I've done that when you say tree.traverse, I expect the tree to contain Option-s and expect an Option[Tree] to be returned. Because, you know... Trees are Foldables :sweat_smile:

How about `findPath(key)`, and having the terminal node be the head of the path instead of duplicated in the tuple?